apps/aecore/src/aec_sync.erl

```diff
@@ -431,64 +555,65 @@ agree_on_height(Uri, _, RH, LHeader, LH, Max, Min) when RH =/= LH ->
          {error, Reason} ->
              lager:debug("Fetching header ~p from ~p failed: ~p", [LH, Uri, Reason]),
              Min
-    end.   
+    end.
 
 do_server_get_missing(Uri) ->
     aec_events:publish(chain_sync, {server_start, Uri}),
     do_get_missing_blocks(Uri),
     aec_events:publish(chain_sync, {server_done, Uri}).
 
 
-fetch_chain(Uri, FromHeight, ToHeight, _) when FromHeight == ToHeight ->
-    aec_events:publish(chain_sync, {client_done, Uri});
-fetch_chain(Uri, FromH, ToH, HashFromH) when FromH < ToH ->
-    case aeu_requests:get_header_by_height(Uri, FromH + 1) of
-        {ok, Header} ->
+fill_pool(Uri, RemoteHeight, AgreedHeight, AgreedHash) ->
+    %% This should change to "get next ?MAX_HEADERS_PER_CHUNK blocks
+    %% starting from provided hash.
+    Chunk = min(RemoteHeight - AgreedHeight, ?MAX_HEADERS_PER_CHUNK),
+    case aeu_requests:get_block_by_height(Uri, AgreedHeight + Chunk) of
+        {ok, Block} ->
```

Comment:
Who checks that the height in the received block is the same height as requested?

Reply:
That's part of the update_from_pool.